tracer/src/Datadog.Trace/LifetimeManager.cs

```diff
@@ -0,0 +1,103 @@
+﻿// <copyright file="LifetimeManager.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace
+{
+    /// <summary>
+    /// Used to run hooks on application shutdown
+    /// </summary>
+    internal class LifetimeManager
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<Tracer>();
+        private static LifetimeManager _instance;
+        private readonly ConcurrentStack<Action> _shutdownHooks = new();
```

Comment:
I don't know if this is an issue, but you're changing the behavior. Previously the callbacks were executed in FIFO order, you're changing that to LIFO.

Reply:
Good point. I'll switch it to a queue to keep the same behaviour